### ISSUE TITLE
Return values & Parameters

### ISSUE THREAD

**Issue by MSP-Greg** [SketchUp/ruby-api-docs#8]
I just noticed that all the return values of methods aren't returning anything other than Object.

It also seems that most of the returns are tagged lower-case.

For instance the method [Sketchup::Model#active_layer](http://sketchup.github.io/ruby-api-docs/Sketchup/Model.html#active_layer-instance_method) shows 'layer' as the return.
If it's changed to 'Layer', does YARD link to the Layer class?

With Ruby code this would be tagged
`@return [Layer] the active layer`

Maybe something is different when using the C++ parser.

**Comment by thomthom** [comment]
This is because the docs never had type notation to it. It would be really great to have this for everything, but 1000+methods actually takes a long time to update.

So it's slated for incremental improvement once we have launched.

**Comment by MSP-Greg** [comment]
> but 1000+methods actually takes a long time to update

Yes, not a trivial task. The kind where you really need a good computer, two or more monitors and a lot of coffee...

Have you considered a (small) community effort?  After all, some people have an interest in good API docs.

FYI, I get 102 classes/modules and 1,242 methods/attributes (`document.querySelectorAll('#full_list a').length`).

**Comment by thomthom** [comment]
> Have you considered a (small) community effort? After all, some people have an interest in good API docs.

I have been wondering if we can pull this off somehow. The problem is that the doc-comments are along with the source code. We can't just make that public. :/ 

Maybe there is a way with a few select under an NDA... Need to look into that.

**Comment by MSP-Greg** [comment]
small <-> select

My thoughts exactly, I didn't mention NDA, but I assumed that would be needed.

FYI, I'm doing some re-working of yard-t2 to make it easy to add a custom header and customize the styling.  Glad I'm doing it, as I'm finding a few of the css things that are font dependent (the custom version for SU is using 'Open Sans' and 'Roboto Mono'.  I can't deal with the red, I'm using 'greyish' highlighting (backgrounds) and a 'bluer' font color for links...
